docs/site/Repositories.md

```diff
@@ -307,50 +307,41 @@ We can now access key-value stores such as [Redis](https://redis.io/) using the
 
 ### Define a KeyValue Datasource
 
-We first need to define a datasource to configure the key-value store. For
-better flexibility, we split the datasource definition into two files. The json
-file captures the configuration properties and it can be possibly overridden by
-dependency injection.
-
-1. redis.datasource.config.json
-
-   ```json
-   {
-     "name": "redis",
-     "connector": "kv-redis",
-     "host": "127.0.0.1",
-     "port": 6379,
-     "password": "",
-     "db": 0
-   }
-   ```
-
-2. redis.datasource.ts
-
-   The class uses a configuration object to set up a datasource for the Redis
-   instance. By default, the configuration is loaded from
-   `redis.datasource.config.json`. We can override it by binding a new object to
-   `datasources.config.redis` for a context.
-
-   ```ts
-   import {inject} from '@loopback/core';
-   import {juggler, AnyObject} from '@loopback/repository';
-   import config from './redis.datasource.config.json';
-
-   export class RedisDataSource extends juggler.DataSource {
-     static dataSourceName = 'redis';
-
-     constructor(
-       @inject('datasources.config.redis', {optional: true})
-       dsConfig: AnyObject = config,
-     ) {
-       super(dsConfig);
-     }
-   }
-   ```
-
-   To generate the datasource automatically, use `lb4 datasource` command and
-   select `Redis key-value connector (supported by StrongLoop)`.
+We first need to define a datasource to configure the key-value store. The
+easiest option is to to generate the datasource automatically using
+`lb4 datasource` command and selecting
+`Redis key-value connector (supported by StrongLoop)`.
+
+The generated datasource file has two parts:
+
+1. A config object with the configuration options provided via CLI.
+2. A DataSource class accepting the configuration via Dependency Injection, with
+   a fall-back to the default config.
+
+```ts
+import {inject} from '@loopback/core';
+import {juggler, AnyObject} from '@loopback/repository';
+
+const config = {
+  name: 'redis',
+  connector: 'kv-redis',
+  host: '127.0.0.1',
+  port: 6379,
+  password: '',
+  db: 0,
+};
+
+export class RedisDataSource extends juggler.DataSource {
+  static dataSourceName = 'redis';
+
```

Comment:
Missing `static defaultConfig = config;`?

Reply:
Fixed in 8c73f9d